open_spiel/games/chess/chess_board.h

```diff
@@ -223,9 +225,9 @@ bool IsMoveCharacter(char c);
 
 std::pair<std::string, std::string> SplitAnnotations(const std::string& move);
 
-template <uint32_t kBoardSize>
+template <uint32_t kBoardSize, bool kAllowCheck>
 class ChessBoard {
- public:
+public:
   ChessBoard();
```

Comment:
I checked the code and maybe I am missing something, but is there a reason to template `kAllowCheck` ? I think it should be sufficient to just pass it as a boolean flag in the constructor. I believe the reason why the board size is templated is because it can then allocate appropriate array sizes on the stack.